### DISCUSSION TITLE
Displaying the requests from the cache and the usage of upstream DNS servers

### DISCUSSION THREAD

**Discussion by lasseveenliese** [AdguardTeam/AdGuardHome#6136] [Ideas]
Hello, thank you very much for AdGuard Home. It's truly a great tool! In Pi-Hole, there's a graph on the dashboard that shows which DNS servers responded to the queries. I would find this feature very useful in AdGuard Home as well. Additionally, it would be helpful in the Query Log if the requests answered from the cache were more prominently visible. In the iOS app, they are very clearly visible and even marked in blue. I would appreciate having these features in the web dashboard as well. Thank you!

**Comment by ainar-g** [comment]
See #1453.  You can already have the upstreams part in the latest Beta, [v0.108.0-b.44](https://github.com/AdguardTeam/AdGuardHome/releases/tag/v0.108.0-b.44).

**Comment by lasseveenliese** [comment]
Thank you! Here's an example of how it could look to display the cache status. Maybe you can consider adding a feature like this?

![image](https://github.com/AdguardTeam/AdGuardHome/assets/129764721/72608aa4-35aa-4f7a-8f29-b46f9715c931)

**Reply by lutfor-diu** [reply]
good to have this :)
